### PR TITLE
OJ-1956: Add error handling to `post-ivq-answers`

### PR DESCRIPTION
**Pull request by surajkumar**
### What changed
Added error handling to the `post-ivq-answers` state machine.
try-catch-throw to `SubmitAnswerHandler` so we can handle HMRC errors 
Updated `post-answers` state machine so that if `post-ivq-answers` fails, it is handled

### Issue tracking

- [OJ-1956](https://govukverify.atlassian.net/browse/OJ-1956)

### Environment variables or secrets

- [X] No environment variables or secrets were added or changed

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-1956]: https://govukverify.atlassian.net/browse/OJ-1956?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: step-functions/post-ivq-answers.asl.json [hunk 1]
```diff
@@ -63,24 +63,25 @@
...
-      "End": true
+      "End": true,
+      "Result": {}
     },
-    "GetParameters": {
+    "Fetch API URL and User Agent": {
       "Type": "Task",
       "Parameters": {
-        "Names": ["${AnswersUrlName}", "${UserAgentName}"]
+        "Names": ["/suraj-kbv-hmrc/AnswersUrl", "/suraj-kbv-hmrc/UserAgent"]
```
Comment (jkunle) [inline step-functions/post-ivq-answers.asl.json:74]: `suraj-kbv` Is this right?

File: step-functions/post-ivq-answers.asl.json [hunk 2]
```diff
@@ -63,24 +63,25 @@
       },
       "ResultPath": "$.oAuthToken"
     },
-    "Do Nothing": {
+    "Nothing left to do": {
       "Type": "Pass",
-      "End": true
+      "End": true,
+      "Result": {}
```
Comment (jkunle) [inline step-functions/post-ivq-answers.asl.json:69]: Is this significant?

Comment (surajkumar) [reply step-functions/post-ivq-answers.asl.json:69]: Yeah it's so the result of the state machine is `{}` and not the unneeded meta data from the other states

File: step-functions/post-ivq-answers.asl.json (hunk 1 above)
Comment (surajkumar) [reply step-functions/post-ivq-answers.asl.json]: Good spot!

**Review by rahulGDS — APPROVED** [review]

**Review by rahulGDS — APPROVED** [review]

**Review by jkunle — APPROVED** [review]
Still don't like having code changes, in this I would have preferred a separate PR one for addressing only the step function changes you have here. And another for the code adding error as well as test. Someone has to write these test and is not usually the person that wrote the code.

**Pull request merged**